### PR TITLE
[Device management] Rename a session (PSG-747)

### LINKED ISSUE #7158: [Device management] Rename a session
#### What would you like to do?
Add the ability to rename a session in the new device management screens (only in session overview screen for now).

#### Why would you like to do it?
User should be able to rename their sessions.

#### How would you like to achieve it?
https://www.figma.com/file/jI0ZdCeKKa51BDORmXdnRU/Settings%3A-UI-Refresh?node-id=976%3A69788

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### Are you willing to provide a PR?

Yes

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Adding a new screen to rename a given session. It can be reached from the session overview screen using the overflow menu.

## Motivation and context

Closes #7158 

## Tests

- Enable the new device manager feature flag
- Go to Settings -> Security & Privacy -> Show all sessions (V2, WIP)
- Press a session entry to go to a session overview
- Press the overflow menu
- Press the rename action
- Check the focus is on the text field and the keaboard is opened when landing on the screen
- Edit the name
- Press the "Save" button
- Check the session name is correctly updated everywhere
- Press a session entry to go to a session overview
- Press the overflow menu
- Press the rename action
- Edit the name
- Press the close button
- Check the session name has not been changed

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Review by jmartinesp — APPROVED** [review]
Minor comment, other than that it LGTM.

File: vector/src/main/java/im/vector/app/features/settings/devices/v2/rename/RenameSessionFragment.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        }
+    }
+
+    override fun invalidate() = withState(viewModel) { state ->
+        if (renameEditTextInitialized.not()) {
+            views.renameSessionEditText.setText(state.editedDeviceName)
+            views.renameSessionEditText.setSelection(views.renameSessionEditText.length())
+            renameEditTextInitialized = true
+        }
+        views.renameSessionSave.isEnabled = state.editedDeviceName.isNotEmpty()
```
[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/rename/RenameSessionFragment.kt:94]: For me it should be in the `ViewState`. The reason is that we should be able to reconstruct the view at any moment using the `ViewState`. So for the enable status of the save button, we should use info from the `ViewState`. Like you said `ViewEvent` is a one time event and so not persisted, meaning we will lose the enable status of the button.
But I don't understand the last part of your comment: 

> could we check the contents of renameSessionEditText before applying these changes instead?

Can you add more details about your suggestion?

[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/rename/RenameSessionFragment.kt:94]: Oh sorry I got it! I will try to improve. I completely misunderstood your suggestion at first...

[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/settings/devices/v2/rename/RenameSessionFragment.kt]: I have done an update: see [a8d7723](https://github.com/vector-im/element-android/pull/7189/commits/a8d7723db02f8760308765e98e7312ba43121cdb). It adds more complexity to the `ViewModel` but I think it more logical this way. Let me know what you think about it.

**Pull request merged**